### PR TITLE
Feat/glu variants

### LINKED ISSUE #959: GLU Variants Improve Transformer
**Is your feature request related to a current problem? Please describe.**
[GLU Variants Improve Transformer](https://arxiv.org/abs/2002.05202)
[labml.ai implementation](https://nn.labml.ai/transformers/glu_variants/simple.html)

**Describe proposed solution**
I thought these would be worth adding since I'm adding other activations. 
I can work on it once #954 is merged.

### PR DESCRIPTION
**Pull request by gdevos010**
Fixes #959

### Summary

Added a components folder to store the feedforward layer and the glu variants. These FFN can be used for any transformer model, i.e., when informer and autoformer are added. 

### Other Information

Is there any feedback on how to use these in other non-transformer based models?

### CODE REVIEW

File: darts/models/components/feed_forward.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+where $W_1$, $W_2$, $b_1$ and $b_2$ are learnable parameters.
+
+Sometimes the
+GELU (Gaussian Error Linear Unit) activation is also used instead of ReLU.
+$$x \Phi(x)$$ where $\Phi(x) = P(X \le x), X \sim \mathcal{N}(0,1)$
+### Gated Linear Units
+
+This is a generic implementation that supports different variants including
+[Gated Linear Units](https://papers.labml.ai/paper/2002.05202) (GLU).
+We have also implemented experiments on these:
```
Comment (hrzn) [inline darts/models/components/feed_forward.py:53]: I'd remove this line and the last two from the docstring.

File: darts/models/forecasting/transformer_model.py [hunk 2]
```diff
@@ -237,7 +252,9 @@ def __init__(
         dropout
             Fraction of neurons affected by Dropout (default=0.1).
         activation
-            The activation function of encoder/decoder intermediate layer, 'relu' or 'gelu' (default='relu').
+            The activation function of encoder/decoder intermediate layer, (default='relu').
+            can be one of the glu variant's FFN ["GLU", "Bilinear", "ReGLU", "GEGLU", "SwiGLU", "ReLU", "GELU"]
```
Comment (hrzn) [inline darts/models/forecasting/transformer_model.py:256]: Can you maybe refer to the paper here, and explain in a couple of sentences what is FFN?

Also: how about optionally adding these to the TFT model which is also transformer based?

File: darts/tests/models/forecasting/test_transformer_model.py [hunk 3]
```diff
@@ -110,3 +110,22 @@ def helper_test_pred_length(self, pytorch_model, series):
         def test_pred_length(self):
             series = tg.linear_timeseries(length=100)
             self.helper_test_pred_length(TransformerModel, series)
+
```
Comment (hrzn) [inline darts/tests/models/forecasting/test_transformer_model.py:113]: Could you also add some "dummy" tests instantiating all of the glu variants? The current PR decreases the code coverage of unit tests (🙀 ), I suspect it comes from the fact that most of these classes are not instantiated in the tests.

**Review by hrzn — COMMENTED** [review]
Looks nice, thanks. I've got a few minor comments.
Out of curiosity: did you run experiments with these variants? Did you observe improvements?

File: darts/models/forecasting/transformer_model.py (hunk 2 above)
Comment (gdevos010) [reply darts/models/forecasting/transformer_model.py]: done

File: darts/tests/models/forecasting/test_transformer_model.py (hunk 3 above)
Comment (gdevos010) [reply darts/tests/models/forecasting/test_transformer_model.py:113]: added test

**Review by hrzn — APPROVED** [review]
LGTM, thanks!
You mentioned working on another model - let us know if that is something you'd see fitting into Darts :)

**Comment by hrzn** [comment]
> @hrzn can we merge it to main?

Yes.

**Pull request merged**